odl/tomo/util/utility.py

```diff
@@ -172,6 +172,142 @@ def axis_rotation_matrix(axis, angle):
     return cos_ang * id_mat + (1. - cos_ang) * dy_mat + sin_ang * cross_mat
 
 
+def rotation_matrix_from_to(from_vec, to_vec):
+    """Return a matrix that rotates ``from_vec`` to ``to_vec`` in 3d.
+
+    Since a rotation from one vector to another has (at least) one
+    degree of freedom, this function makes deliberate but still
+    arbitrary choices to fix these free parameters. See Notes for
+    details.
+
+    Parameters
+    ----------
+    from_vec, to_vec : `array-like`
+        Vectors between which the returned matrix rotates. They should not
+        be very close to zero or collinear. Their shapes must be equal
+        to ``(3,)``.
+
+    Returns
+    -------
+    matrix : `numpy.ndarray`, shape ``(3, 3)``
+        A matrix rotating ``from_vec`` to ``to_vec``. Note that the
+        matrix does *not* include scaling, i.e. it is not guaranteed
+        that ``matrix.dot(from_vec) == to_vec``.
+
+    Examples
+    --------
+    In two dimensions, rotation is simple:
+
+    >>> from_vec, to_vec = [1, 0], [1, 1]
+    >>> mat = rotation_matrix_from_to(from_vec, to_vec)
+    >>> to_vec_normalized = np.divide(to_vec, np.linalg.norm(to_vec))
+    >>> np.allclose(mat.dot([1, 0]), to_vec_normalized)
+    True
+    >>> from_vec, to_vec = [1, 0], [-1, 1]
+    >>> mat = rotation_matrix_from_to(from_vec, to_vec)
+    >>> to_vec_normalized = np.divide(to_vec, np.linalg.norm(to_vec))
+    >>> np.allclose(mat.dot([1, 0]), to_vec_normalized)
+    True
+
+    Rotation in 3d by less than ``pi``:
+
+    >>> from_vec, to_vec = [1, 0, 0], [-1, 1, 0]
+    >>> mat = rotation_matrix_from_to(from_vec, to_vec)
+    >>> to_vec_normalized = np.divide(to_vec, np.linalg.norm(to_vec))
+    >>> np.allclose(mat.dot([1, 0, 0]), to_vec_normalized)
+    True
+
+    Rotation by more than ``pi``:
+
+    >>> from_vec, to_vec = [1, 0, 0], [-1, -1, 0]
+    >>> mat = rotation_matrix_from_to(from_vec, to_vec)
+    >>> to_vec_normalized = np.divide(to_vec, np.linalg.norm(to_vec))
+    >>> np.allclose(mat.dot([1, 0, 0]), to_vec_normalized)
+    True
+
+    Notes
+    -----
+    In 3d, the matrix corresponds to a rotation around the normal vector
```

Comment:
I added a link to Rodrigues' formula in the documentation. I'm not aware of a more specific name for this kind of matrix.